pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun_test.go

```diff
@@ -0,0 +1,626 @@
+			if err != nil {
+				t.Fatal(err)
+			}
+			errs := verifyTaskOCIBundle(ctx, tc.bundle, sv, signature, k8sclient)
+			if (errs != nil) != tc.wantErr {
+				t.Fatalf("verifyResources get err %v, wantErr %t", err, tc.wantErr)
+			}
+		})
+	}
+
+}
+
+func TestGetDigest(t *testing.T) {
+	ctx := context.Background()
+	// Create registry server
+	s := httptest.NewServer(registry.New())
+	defer s.Close()
+	u, _ := url.Parse(s.URL)
+
+	// Push OCI bundle
+	dig, err := pushOCIImage(t, u, ts)
+	t.Logf("Digest: %v", dig.String())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kc, err := k8schain.New(ctx, k8sclient, k8schain.Options{
+		Namespace:          nameSpace,
+		ServiceAccountName: serviceAccount,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tcs := []struct {
+		name     string
+		imageRef string
+		wantErr  bool
+	}{
+		{
+			name:     "OCIBundle Pass Verification",
+			imageRef: u.Host + "/task/" + ts.Name,
+			wantErr:  false,
+		},
+		{
+			name:     "OCIBundle Fail Verification with empty signature",
+			imageRef: u.Host + "/task/" + tsTampered.Name,
+			wantErr:  true,
+		},
+		{
+			name:     "OCIBundle Fail Verification with empty Bundle",
+			imageRef: "",
+			wantErr:  true,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err = getDigest(ctx, tc.imageRef, kc)
+			if (err != nil) != tc.wantErr {
+				t.Fatalf("verifyResources get err %v, wantErr %t", err, tc.wantErr)
+			}
+		})
+	}
+
+}
+
+// Generate key files to tmpdir, set configMap and return signer
+func getSingerFromFile(t *testing.T, ctx context.Context) (signature.Signer, error) {
+	tmpDir := t.TempDir()
+	privateKeyPath, _ := generateKeyFile(t, tmpDir, pass(password))
+	signer, err := cosignsignature.SignerFromKeyRef(ctx, privateKeyPath, pass(password))
+	if err != nil {
+		t.Fatal(err)
+	}
+	signgingConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: nameSpace,
+			Name:      signgingConfigMap,
+		},
+		Data: map[string]string{"path": tmpDir},
+	}
+
+	_, err = k8sclient.CoreV1().ConfigMaps(system.Namespace()).Create(ctx, signgingConfigMap, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return signer, nil
+}
+
+func pushOCIImage(t *testing.T, u *url.URL, task *v1beta1.Task) (v1types.Hash, error) {
+	ref, err := remotetest.CreateImage(u.Host+"/task/"+task.Name, task)
+	if err != nil {
+		t.Errorf("uploading image failed unexpectedly with an error: %v", err)
+	}
+
+	imgRef, err := imgname.ParseReference(ref)
+	if err != nil {
+		t.Errorf("digest %s is not a valid reference: %v", ref, err)
+	}
+
+	img, err := remote.Image(imgRef, remote.WithAuthFromKeychain(authn.DefaultKeychain))
+	if err != nil {
+		t.Errorf("could not fetch created image: %v", err)
+	}
+
+	dig, err := img.Digest()
+	if err != nil {
+		t.Errorf("failed to fetch img manifest: %v", err)
+	}
+	return dig, nil
+}
+
+func getKeysSVRaw(t *testing.T, pf cosign.PassFunc) (*cosign.KeysBytes, signature.SignerVerifier, error) {
```

Comment:
```suggestion
func getKeysSVRaw(t *testing.T, pf cosign.PassFunc) (signature.SignerVerifier, error) {
```

I don't think this is ever used outside this func.

Reply:
ah yes, thanks!